resolver-dns/src/main/java/io/netty/resolver/dns/DnsResolveContext.java

```diff
@@ -301,20 +300,24 @@ private String cnameResolveFromCache(String name) {
     }
 
     private String cnameResolveFromCacheLoop(DnsCnameCache cnameCache, String first, String mapping) {
-        // Detect loops using a HashSet. We use this as last resort implementation to reduce allocations in the most
-        // common cases.
-        Set<String> cnames = new HashSet<String>(4);
-        cnames.add(first);
-        cnames.add(mapping);
+        // Detect loops by advance only every other iteration. This is the same concept as used in guava
```

Comment:
May way to reference this (instead of or in addition to Guava):
https://en.wikipedia.org/wiki/Cycle_detection#Floyd's_Tortoise_and_Hare